File: app/src/main/java/com/example/androidbasic/ui/activity/BaseActivity.java
package com.example.androidbasic.ui.activity;

import android.support.v7.app.AlertDialog;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;

/**
 * Created by vikas on 26/9/16.
 */

public class BaseActivity extends AppCompatActivity {

    private String LOG_TAG;

    public void setLogTag(String tag) {
        LOG_TAG = tag;
    }

    public void printLog(String string) {
        Log.d(LOG_TAG,string);
    }

    public void showDialog() {
        AlertDialog.Builder builder = new AlertDialog.Builder(this);
        builder.setMessage("Dialog came up the screen");
        builder.setPositiveButton("Close",null);
        builder.show();
    }
}

File: app/src/main/java/in/vikasrathod/androidbasic/ui/activity/BaseActivity.kt
package `in`.vikasrathod.androidbasic.ui.activity

import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import android.util.Log

/**
 * Created by vikas on 26/9/16.
 */

open class BaseActivity : AppCompatActivity() {

    private var LOG_TAG: String? = null

    fun setLogTag(tag: String) {
        LOG_TAG = tag
    }

    fun printLog(string: String) {
        Log.d(LOG_TAG, string)
    }

    fun showDialog() {
        val builder = AlertDialog.Builder(this)
        builder.setMessage("Dialog came up the screen")
        builder.setPositiveButton("Close", null)
        builder.show()
    }
}
